server/src/main/kotlin/fi/oph/ludos/s3/S3Service.kt

```diff
@@ -51,4 +51,10 @@ class S3Service(val s3: S3Client) {
             throw ex
         }
     }
+
+    fun deleteObject(key: String) {
+        val objectRequest = DeleteObjectRequest.builder().bucket(bucket).key(key).build()
+
+        s3.deleteObject(objectRequest)
+    }
```

Comment:
s3.deleteObjectin SdkExceptionin vois nappaa kii ja lokittaa errorin ja heittää poikkeuksen eteenpäin kuten getObjectissa. Saman muutoksen vois tehdä myös putObjectille. Vois helpottaa debuggailuja joskus. (Ja lisäks metodien kutsujan pitää myös muistaa ottaa SdkException kiinni.)

Reply:
Näin yleisesti tän S3Servicen olemassaolon tarkoitus vähän mietityttää. Nyt tän pointtina näyttäis olevan varmistaa käynnistyksessä, että S3-yhteys pelaa ja lisäksi toimia hyvin ohuena wrapperina S3Clientille. S3Servicen käyttäjän pitää nyt tietää aika paljon tuon S3Clientin rajapinnan detaileista, kuten pitää osata ottaa kiinni SdkException, ja getObjectin käyttäjän pitää osata käpistellä `ResponseInputStream<GetObjectResponse>`.

En osaa sanoa, onko järkeä abstrahoida tuota S3Clientia yhtään enempää, eli ehkäpä just tällainen moduuli palvelee meitä parhaiten just nyt. Eli tää on oikeastaan jonkinlainen s3-helpperi, joka vähän helpottaa yleisiä käyttötapauksia.

Ehdotan, että nimi vaihdetaan olemaan `S3Helper`, ja annotaatio vaihdetaan `@Service` -> `@Component`, kun tuntuu vähän väärältä kutsua Service-tyyppistä asiaa repository-tasolta. (Toivottavasti myös `@Component` alustetaan, kun järjestelmä käynnistyy, jotta S3-yhteystsekki jatkaa toimintaansa).

(Jos tää abstrahoisi S3Clientin yksityiskohdat kunnolla, niin sit oikea nimi ois ehkä S3Repository, kun tää tekee just tallentamiseen ja tiedonhakuun liittyviä asioita :thinking: )